components/scream/src/dynamics/homme/physics_dynamics_remapper.hpp

```diff
@@ -695,7 +695,11 @@ do_remap_fwd() const
 
   const auto concurrency = KT::ExeSpace::concurrency();
 #ifdef KOKKOS_ENABLE_CUDA
+#ifdef KOKKOS_ENABLE_DEBUG
```

Comment:
OK, but none of this special-case code should be here. That's not the subject of this PR, though.

I'll just note that this code is working around the fact that EKAT's get_default_team_policy is currently specific to the SCREAM physics. That should be moved to SCREAM's physics directory's utilities. EKAT itself should provide basic utitlies like max team size. Then SCREAM can use that to develop a few basic team policies for different parts of the computation.

Reply:
@tcclevenger it looks like you introduce the original code. At some point I recommend you review https://github.com/E3SM-Project/E3SM/blob/master/components/homme/src/share/cxx/ExecSpaceDefs.hpp and its associated cpp file, as well as the similar code in https://github.com/E3SM-Project/EKAT/blob/master/src/ekat/kokkos/ekat_kokkos_utils.hpp, to work out a coherent approach to making TeamPolicys. EKAT should focus on providing basic utils like max team size as a function of debug build flag. SCREAM should use these utils to develop a set of TeamPolicy constructors for each use case. App code like this should never have KOKKOS_ENABLE_CUDA and similar details.

Reply:
@ambrad Ok, I'll look into a better way for defining these things. And then implement in scream.